clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -11231,8 +11231,12 @@ void LinkerWrapper::ConstructJob(Compilation &C, const JobAction &JA,
 
     bool IsUsingLTO = D.isUsingLTO(/*IsDeviceOffloadAction=*/true);
     auto LTOMode = D.getLTOMode(/*IsDeviceOffloadAction=*/true);
-    if (IsUsingLTO && LTOMode == LTOK_Thin)
+    if (IsUsingLTO && LTOMode == LTOK_Thin) {
       CmdArgs.push_back(Args.MakeArgString("-sycl-thin-lto"));
+      // TODO: Pass the same value for this argument once we start using it
+      // for non-thinLTO.
+      CmdArgs.push_back(Args.MakeArgString("-sycl-module-split-mode=auto"));
```

Comment:
I looked at the logic surounding 'sycl-module-split-mode' option. If this option is set (to any value), we use the SYCL split library. Otherwise, we call the SYCL post link tool. Given the current state of affairs, i would suggest that we do not set this option.
Thanks